social_protection/validation.py

```diff
@@ -7,12 +11,59 @@ class BenefitPlanValidation(BaseModelValidation):
 
     @classmethod
     def validate_create(cls, user, **data):
+        incoming_code = data.get('code')
+        code_error = check_bf_unique_code(incoming_code)
+        if code_error:
+            raise ValidationError(code_error[0]['message'])
+        incoming_name = data.get('name')
+        name_error = check_bf_unique_name(incoming_name)
+        if name_error:
+            raise ValidationError(name_error[0]['message'])
+        incoming_schema = data.get('beneficiary_data_schema')
+        schema_error = is_valid_json_schema(incoming_schema)
+        if schema_error:
+            raise ValidationError(schema_error[0]['message'])
         super().validate_create(user, **data)
 
     @classmethod
     def validate_update(cls, user, **data):
+        uuid = data.get('id')
+        incoming_code = data.get('code')
+        code_error = check_bf_unique_code(incoming_code, uuid)
+        if code_error:
+            raise ValidationError(code_error[0]['message'])
+        incoming_name = data.get('name')
+        name_error = check_bf_unique_name(incoming_name, uuid)
+        if name_error:
+            raise ValidationError(name_error[0]['message'])
+        incoming_schema = data.get('beneficiary_data_schema')
+        schema_error = is_valid_json_schema(incoming_schema)
+        if schema_error:
+            raise ValidationError(schema_error[0]['message'])
         super().validate_update(user, **data)
 
     @classmethod
     def validate_delete(cls, user, **data):
         super().validate_delete(user, **data)
+
+
+def check_bf_unique_code(code, uuid=None):
+    instance = BenefitPlan.objects.filter(code=code, is_deleted=False).first()
+    if instance and instance.uuid != uuid:
+        return [{"message": "BenefitPlan code %s already exists" % code}]
```

Comment:
Please wrap message code in _(translation).

Reply:
I'm also not entirely sure about this return formats. I know that we're using them as they're compatible with gql error handling approach but on the same time it doesn't seem right for function called `check_bf_unique` code, which is supposed to check if code is unique (return true/false or raise exception if it isn't). With this output we're actually returning list of validation issues for the bf unique code. Do you have any ideas on how to handle it? We could simply rename method to `validate_bf_code` and add validation for empty/blank values.

Reply:
done